client_impl/gitlab_client.py

```diff
@@ -0,0 +1,104 @@
+from datetime import datetime
+from typing import Optional, List
+
+import gitlab
+import gitlab.v4.objects.projects
+from gitlab.v4.objects import CurrentUser, ProjectCommit
+
+from abstractions import ScmClient, GitRepository, GitCommit, PullRequest
+
+
+class GitLabClient(ScmClient):
+
+    def __init__(self, pat: str, api_base_url: Optional[str] = None):
+        super().__init__(pat, api_base_url)
+        self._gitlab_client = gitlab.Gitlab(api_base_url, private_token=pat) if api_base_url else gitlab.Gitlab(
+            private_token=pat)
+        self._gitlab_client.auth()
```

Comment:
I propose to move the `auth()` call to `get_username()`, because it is uncommon to make network calls in a constructor